.github/workflows/python.yaml

```diff
@@ -6,85 +6,226 @@ on:
       - main
   pull_request:
 
-
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+  PACKAGE_NAME: graphlib2
+  PYTHON_VERSION: "3.7" # to build abi3 wheels
+
 jobs:
-  sdist:
-    name: Build sdist
-    runs-on: ubuntu-latest
+  macos:
+    runs-on: macos-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
-        name: Install Python
-        with:
-          python-version: '3.7'
-      - name: Install deps
-        run: pip install -U maturin
-      - name: Build sdist
-        run: maturin build --release --strip
-      - uses: actions/upload-artifact@v2
-        with:
-          path: ./target/wheels/*.tar.gz
-          name: dist
-  build_wheels:
-    name: Build wheels on ${{ matrix.os }} / Python ${{ matrix.python }}
-    runs-on: ${{ matrix.os }}
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          architecture: x64
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          default: true
+      - name: Build wheels - x86_64
+        uses: messense/maturin-action@v1
+        with:
+          target: x86_64
+          args: --release --out dist
+      - name: Install built wheel - x86_64
+        run: |
+          pip install dist/${{ env.PACKAGE_NAME }}-*.whl --force-reinstall
+          pip install pytest
+          pytest -v
+      - name: Build wheels - universal2
+        uses: messense/maturin-action@v1
+        with:
+          args: --release --universal2 --out dist --no-sdist
+      - name: Install built wheel - universal2
+        run: |
+          pip install dist/${{ env.PACKAGE_NAME }}-*.whl --force-reinstall
```

Comment:
```suggestion
          pip install dist/${{ env.PACKAGE_NAME }}-*universal2.whl --force-reinstall
```

Reply:
Thanks! I had to do it this way because this lib depends on `typing_extensions`, so it needs access to PyPi, making `--no-index` a no go.